### PR TITLE
Fix zephyr getting started

### PR DESCRIPTION
**Pull request by szczys**
Flesh out the Zephyr section of the Getting Started guide

* Add summary
* Make the Zephyr install board-agnostic
* Move to using the Golioth Firmware SDK for the install
* Add page detailing building/running samples

This covers mainline Zephyr. An additional section will be added for NCS in a future PR.

### CODE REVIEW

File: docs/getting-started/3-device-examples/2-compile-example-code/1-zephyr/01-set-up-zephyr.md [hunk 1]
```diff
@@ -61,26 +57,19 @@ Review your vendor's board support in the [Zephyr Board Index](https://docs.zeph
...
+### Build Firmware
 
 Your system is all set up and ready to start building & flashing with Zephyr.
-Verify by building a minimal sample for the NXP mimxrt1060_evkb
+Verify by building a minimal sample from the Zephyr tree.
 
 import BuildSample from '/docs/partials-common/sample-build.mdx'
 
-<BuildSample board="mimxrt1060_evk"/>
+<BuildSample board="your_board_name"/>
```
Comment (sam-golioth) [inline docs/getting-started/3-device-examples/2-compile-example-code/1-zephyr/01-set-up-zephyr.md:67]: Nice! Love that we're making this more generic

File: docs/partials-common/zephyr-install-golioth-firmware-sdk-for-windows.md [hunk 2]
```diff
@@ -5,8 +5,7 @@ import TabItem from '@theme/TabItem';
 
     ```shell
     cd c:\
-    west init -m https://github.com/golioth/golioth-zephyr-sdk.git --mf west-zephyr.yml golioth-zephyr-workspace
-    cd golioth-zephyr-workspace
+    west init -m https://github.com/golioth/golioth-firmware-sdk.git --mf west-zephyr.yml golioth-zephyr-workspace
```
Comment (sam-golioth) [inline docs/partials-common/zephyr-install-golioth-firmware-sdk-for-windows.md:8]: Should we add `--mr v0.9.0` ?

**Review by sam-golioth — APPROVED** [review]
Looks good!

File: docs/partials-common/zephyr-install-golioth-firmware-sdk-for-windows.md (hunk 2 above)
Comment (szczys) [reply docs/partials-common/zephyr-install-golioth-firmware-sdk-for-windows.md:8]: Ah, yes. Good catch! I'll add this to both partials. Thanks.

**Pull request merged**